src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,146 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "validLogin";
+    private static final String VALID_PASSWORD = "validPassword";
+    private static final String EXISTING_USER_LOGIN = "existingUserLogin";
+    private static final String SHORT_LOGIN = "log";
+    private static final String SHORT_PASSWORD = "pass";
+    private static final String EMPTY_STRING = "";
+    private static final int VALID_AGE = 24;
+    private static final int INVALID_AGE = 15;
+
+    private RegistrationService registrationValidator;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        registrationValidator = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    public void register_existingUser_notOk() {
+        User existingUser = createUser(EXISTING_USER_LOGIN, VALID_PASSWORD, VALID_AGE);
+        storageDao.add(existingUser);
+        assertThrows(RegistrationException.class, ()
+                -> registrationValidator.register(existingUser));
+    }
+
+    @Test
+    public void register_newUser_ok() throws RegistrationException {
+        User newUser = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        registrationValidator.register(newUser);
+        assertEquals(newUser, storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_validLogin_ok() throws RegistrationException {
+        User newUser = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        registrationValidator.register(newUser);
+        assertEquals(newUser, storageDao.get(VALID_LOGIN));
+    }
```

Comment:
The test `register_validLogin_ok` seems redundant as it is testing the same scenario as `register_newUser_ok`. Consider removing this test or ensuring that each test case tests a unique scenario.